### ISSUE TITLE
Multiple self-referencing foreign keys: Unable to determine the relationship represented by navigation property. -5.0.0-preview.7

### ISSUE THREAD

**Issue by qsdfplkj** [dotnet/efcore#21968]
Creating a migration for the following entity:

    public class Person
    {
        public long Id { get; set; }

        public string Name { get; set; }

        public long? MotherId { get; set; }

        [ForeignKey(nameof(MotherId))]
        public Person Mother { get; set; }

        public long? FatherId { get; set; }

        [ForeignKey(nameof(FatherId))]
        public Person Father { get; set; }

        public long? StephMotherId { get; set; }

        [ForeignKey(nameof(StephMotherId))]
        public Person StepMother { get; set; }
    }

### Steps to reproduce

create a migration for the person entity.

This isn't working on 3.1.6 either.

### Further technical details

EF Core version: 5.0.0-preview.7 
Database provider: sqlserver
Target framework: .NET Core 3.1)
Operating system: windows 10 vs 2019. -16.6.5

**Comment by smitpatel** [comment]
You need to configure the relationship manually using Fluent API or InversePropertyAttribute. See [our documentation](https://docs.microsoft.com/ef/core/modeling/relationships?tabs=fluent-api%2Cfluent-api-simple-key%2Csimple-key) for more details.

**Comment by AndriySvyryd** [comment]
**Note to triage:** When `ForeignKeyAttribute` is specified on both navigations we split the relationship into two, therefore there are no ambiguous candidates in this scenario, so we don't need to throw.

**Comment by qsdfplkj** [comment]
@AndriySvyryd It is working for self-referencing foreign keys but adding a third breaks it.

**Comment by AliMahmodi** [comment]
@qsdfplkj  Have you find any solutions for this ?
If you don't mind please share it

**Comment by qsdfplkj** [comment]
I didn't. I tried creating derived types and use some kind of table per hierarchy approach which seemed to work but in the end it we did not use any of it. The example I gave is just a reproduction of an issue I had. In the end our design uses an extra link table that references two (which seem to work) persons. Where each person only has a single parent and references this link table. I no longer know the exact details hope this helps. I would suggest to explore the derived type approach if possible because in our case a person sometimes a mother and sometimes a stepmother.
